### PR TITLE
Clean up `MlsGroup*` builders

### PR DESCRIPTION
**Pull request by kkohbrok**
This PR removes the dedicated `required_capabilities` and `external_senders` fields (along with their getters and setters) in favor of the `group_context_extension` field. This makes processing during group creation easier.

### CODE REVIEW

File: openmls/src/group/mls_group/builder.rs [hunk 1]
```diff
@@ -194,22 +195,6 @@ impl MlsGroupBuilder {
         self
     }
 
-    /// Sets the `external_senders` property of the MlsGroup.
-    pub fn external_senders(mut self, external_senders: ExternalSendersExtension) -> Self {
```
Comment (franziskuskiefer) [inline openmls/src/group/mls_group/builder.rs:198]: Remmoving these public functions should go into the changelog.
We should add the others we added as well.

File: openmls/src/group/tests/external_remove_proposal.rs [hunk 2]
```diff
@@ -30,7 +30,10 @@ fn new_test_group(
     let mls_group_config = MlsGroupCreateConfig::builder()
         .wire_format_policy(wire_format_policy)
         .crypto_config(CryptoConfig::with_default_version(ciphersuite))
-        .external_senders(external_senders)
+        .with_group_context_extensions(Extensions::single(Extension::ExternalSenders(
```
Comment (franziskuskiefer) [inline openmls/src/group/tests/external_remove_proposal.rs:33]: Nothing to change here. But I noticed that we're not consistent using `with_property` or only `property` in builders.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, generally looks good. Let's try to keep the changelog a little up to date.

File: openmls/src/group/tests/external_remove_proposal.rs (hunk 2 above)
Comment (kkohbrok) [reply openmls/src/group/tests/external_remove_proposal.rs:33]: True, but that's a lot of additional changes that I'd rather have in a separate PR. We'll also have to agree on a style. Maybe generally something we should do once we're done with refactoring the builders.

**Review by franziskuskiefer — APPROVED** [review]
Thanks

**Pull request merged**